drivers/gpio/gpio_intel.c

```diff
@@ -569,11 +566,12 @@ int gpio_intel_init(const struct device *dev)
 
 	isr_devs[nr_isr_devs++] = dev;
 
+#ifdef CONFIG_SOC_APOLLO_LAKE
 	/* route to IRQ 14 */
 
 	sys_bitfield_clear_bit(regs(dev) + REG_MISCCFG,
 			       MISCCFG_IRQ_ROUTE_POS);
-
+#endif
```

Comment:
Have you verified that this code block was not needed for EHL?

Btw, whenever possible, it's better to use `if (IS_ENABLED(CONFIG_...)) {` instead of `#ifdef`, since it produces more readable code. I think in this case it should be possible to convert the #ifdef to `IS_ENABLED()`